src/ur_realtime_communication.cpp

```diff
@@ -84,17 +84,25 @@ void UrRealtimeCommunication::halt() {
 	comThread_.join();
 }
 
-void UrRealtimeCommunication::addCommandToQueue(std::string inp) {
+void UrRealtimeCommunication::addCommandToQueue(const std::string& inp) {
 	int bytes_written;
-	if (inp.back() != '\n') {
-		inp.append("\n");
-	}
+  assert( inp.back == '\n');
```

Comment:
good catch with the parentheses.
The main reason here is to be able to pass a const string& 
Furthermore, forgetting the "\n" should be considered a programming error that fails, not a normal behavior that can reasonably happen at runtime.